- avoid (harmless) compiler warning about incompatible function pointer types
- add default_paths to STORE for checking signatures (aka /etc/ssl/cert.pem)
- do not call EVP_add_digest()

Index: source/helpers/pkcs7/pkcs7-openssl.c
--- source/helpers/pkcs7/pkcs7-openssl.c.orig
+++ source/helpers/pkcs7/pkcs7-openssl.c
@@ -203,7 +203,7 @@ static int stream_free(BIO *b)
 	return 1;
 }
 
-static long stream_callback_ctrl(BIO *b, int cmd, bio_info_cb *fp)
+static long stream_callback_ctrl(BIO *b, int cmd, BIO_info_cb *fp)
 {
 	return 1;
 }
@@ -519,6 +519,9 @@ check_certificate(fz_context *ctx, pdf_pkcs7_verifier 
 		}
 	}
 
+
+	/* Add default paths */
+	X509_STORE_set_default_paths(st);
 	res = pk7_verify_cert(st, pk7sig);
 
 exit:
@@ -793,8 +796,10 @@ pdf_pkcs7_signer *pkcs7_openssl_read_pfx(fz_context *c
 
 		OpenSSL_add_all_algorithms();
 
+#ifndef LIBRESSL_VERSION_NUMBER
 		EVP_add_digest(EVP_md5());
 		EVP_add_digest(EVP_sha1());
+#endif
 
 		ERR_load_crypto_strings();
 
